client/src/views/home/DeviceJoinOrganizationModal.vue

```diff
@@ -28,20 +28,35 @@
     <div
       class="modal"
       :class="{
-        wizardTrue: pageStep > DeviceJoinOrganizationStep.Information && pageStep != DeviceJoinOrganizationStep.Finish,
+        wizardTrue: pageStep > DeviceJoinOrganizationStep.Information && pageStep != DeviceJoinOrganizationStep.Finish && isLargeDisplay,
       }"
     >
-      <ion-header class="modal-header">
+      <ion-header
+        class="modal-header"
+        v-if="isLargeDisplay"
+      >
         <ion-title class="modal-header__title title-h2">
-          {{ $msTranslate(getTitleAndSubtitle().title) }}
+          {{ $msTranslate(getStep().title) }}
         </ion-title>
         <ion-text
-          v-if="getTitleAndSubtitle().subtitle"
+          v-if="getStep().subtitle"
           class="modal-header__text body"
         >
-          {{ $msTranslate(getTitleAndSubtitle().subtitle) }}
+          {{ $msTranslate(getStep().subtitle) }}
         </ion-text>
       </ion-header>
+      <small-display-step-modal-header
```

Comment:
Same here, `SmallDisplayStepModalHeader`'s API is incorrect, and the close button should be hidden on the last step.